Building quality apps/Pull requests.md

```diff
@@ -1,19 +1,72 @@
+
+### The algorithm
+
+Duplicate the browser tab which contains the PR. One will be for writing the description. Another will be for going through the 'Files changed'. If you have multiple displays, even better. Put the tab in which you'll write a description on one display, and the tab with the changes on the other. This will make it easier for you to switch between writing a description and looking at the changes.
+
+Go through the files changed one-by-one, and go through the change hunks one-by-one.
+
+That change you're looking at belongs to a group of changes aimed towards a goal. Does your description contain a section for this goal? If not, create a section. Use [markdown](https://guides.github.com/features/mastering-markdown/) to your advantage. Describe the change inside this section. Perhaps it's already described, perhaps you need a single sentence, perhaps you need a paragraph, perhaps you need to reword what you wrote previously.
+
+Not sure what to do with the change you're looking at right now? Skip it and come back to it later.
+
+Finished with all the changes in a file? Collapse the file to get it out of your way and make it easier to navigate and see what's left.
+
+After you're done, give it a last couple of re-reads and re-iterations.
+
+## Finally
+
+After reading your description, the reviewer should be able to have a single pass through the 'Files changed' section, without going back and forth hopelessly trying to make sense of the changes and hopelessly trying to build a mental model.
+
+The title should follow the same format as a commit message heading. That is, use the imperative mood, as if giving a command, and keep it short. You can leave the branch name, if you choose to, but it takes 10 seconds to think of something better. The title is far less important than the description, though.
+
+How much time should you invest in writing a PR description? Use your own reasoning on a case-by-case basis, as with everything. For very small PRs, it will take you a couple of minutes. For a small PR, 15 minutes should be sufficient. For a moderate PR, it's perfectly fine to take 30 minutes. For larger PRs, you may need an hour, or more. The return on investment is huge in either case. The more you practise, the faster and better you'll be, just like with any other skill.
+
+I believe that some examples can be useful. I'll use my own PRs, simply because I was trying to hold myself up to a standard. They are by no means perfect. Form your own opinion about them. Pick up what you like, and change what you don't. Try to review them, as a matter of fact. (Please don't leave actual comments.) See how it feels going through the files changed, or going through the commits. See what made it easier for you, and try to figure out what's still missing from my descriptions. Perhaps try to go through the changes without reading the description, as well.
```

Comment:
The personal note is kinda cool and I like the whole try-it-yourself concept, but we don't usually use the 1st person in this handbook. Maybe there's a way we could rewrite this to be a bit more 3rd person-ish? 😅

Reply:
Yeah, I see. If we're going to have to remove the examples, this paragraph goes away anyway, though.